README.md

```diff
@@ -208,7 +208,31 @@ At a minimum, all Roaring implementations should be able to parse the two files
 
 ## Unsigned integers in Java
 
-
 Java lacks native unsigned integers, but integers are still considered to be unsigned within Roaring, and ordered  according to  ``Integer.compareUnsigned``. 
 
 For integers in [0,2147483647], the unsigned and signed integers are undistinguisable since Java relies a 32-bit two's complement format for its ``int`` type.
+
+# Extention for 64B implementations
+
+Some Roaring bitmap implementations may offer a 64-bits implementation. This section proposes a portable format, compatible with some (but not all) 
```

Comment:
We should specify what the range of values are. For 64-bit integers, there are two possibilities... `[-9223372036854775808, 9223372036854775807]` or `[0,18446744073709551616)`.

Reply:
Roaring64NavigableMap enables both modes, but CRoaring targets only unsigned longs.

Reply:
We may serialization following the format, but it would lead to unexpected results if read from another implementation. Also, the portable format does not add a hint for this option (signed vs unsigned) (while legacy Roaring64NavigableMap serialization format included a boolean to indicate if it handles signed or unsigned longs).

Reply:
When we write out/read the data, do we require that the keys be in increasing order? If so, we must specify the order.

Reply:
Regarding CRoaring: I do not get why we serialize the number of buckets as uint64 instead of uint32. Java legacy serialization format was serializing as unit32 (and its current implementation would not handle a long > `2^32`).

Is it a bug in the design, or is there a real motivation @lemire ?

Reply:
> Is it a bug in the design, or is there a real motivation @lemire ?

I did not design any of these serialization formats. They have grown organically through various contributions. As far as I am concerned, we are even allow to change them. Nobody should rely on an undocumented format for long-term storage.

It is only when we start to ask that the formats be supported and interoperable that we need to think seriously about what we are doing.

Regarding the gap you have identified. Nobody care about the waste of two bytes. In fact, having two zero bytes in the specification is not bad. It can be used to detect mistakes (the input is not really what you think it is).

We may even use these two bytes later for other purposes.